edit-post/components/header/copy-content-menu-item/index.js

```diff
@@ -22,17 +21,9 @@ function CopyContentButton( { editedPostContent, hasCopied, setState } ) {
 	);
 }
 
-const Enhanced = compose(
+export default compose(
 	query( ( select ) => ( {
 		editedPostContent: select( 'core/editor' ).getEditedPostAttribute( 'content' ),
```

Comment:
Maybe we should perform this call only when clicking on the button. Serializing blocks can harm performance.

Reply:
Would you like to call it directly using `wp.data.select` or expose it as a function here?
@mcsf what's your take on this?

Reply:
On a second thought, maybe it's not that severe because it's only shown when you open the dropdown menu, it's probably run only once because you don't edit content when this menu is open.

Reply:
Right, it doesn't render when you edit your post. It only renders once when you open the dropdown.

Reply:
Actually, `ClipboardButton` is flexible and accepts a function for `text`, so the following works:

```diff
diff --git a/edit-post/components/header/copy-content-menu-item/index.js b/edit-post/components/header/copy-content-menu-item/index.js
index 88561ae1..a55688ad 100644
--- a/edit-post/components/header/copy-content-menu-item/index.js
+++ b/edit-post/components/header/copy-content-menu-item/index.js
@@ -2,11 +2,13 @@
  * WordPress dependencies
  */
 import { ClipboardButton, withState } from '@wordpress/components';
-import { compose } from '@wordpress/element';
-import { query } from '@wordpress/data';
+import { select } from '@wordpress/data';
 import { __ } from '@wordpress/i18n';
 
-function CopyContentMenuItem( { editedPostContent, hasCopied, setState } ) {
+const editedPostContent = () =>
+	select( 'core/editor' ).getEditedPostAttribute( 'content' );
+
+function CopyContentMenuItem( { hasCopied, setState } ) {
 	return (
 		<ClipboardButton
 			text={ editedPostContent }
@@ -21,9 +23,6 @@ function CopyContentMenuItem( { editedPostContent, hasCopied, setState } ) {
 	);
 }
 
-export default compose(
-	query( ( select ) => ( {
-		editedPostContent: select( 'core/editor' ).getEditedPostAttribute( 'content' ),
-	} ) ),
-	withState( { hasCopied: false } )
+export default withState(
+	{ hasCopied: false }
 )( CopyContentMenuItem );

```